clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/client/AzureNetworkClient.groovy

```diff
@@ -271,25 +284,179 @@ class AzureNetworkClient extends AzureBaseClient {
   }
 
   /**
-   * Retrieve the App Gateway backend address pool. Create a new one if it doesn't already exist
-   * @param resourceGroupName - Resource Group where desired resources exist
-   * @param appGatewayName - Name of the Application Gateway
-   * @param backendAddressPoolName - name of the backend address pool to locate/create
-   * @param azureAppGateway - Instance of a ApplicationGateway
-   * @return
+   * It retrieves an Azure Application Gateway for a given set of credentials, resource group and name
+   * @param appGatewayDescription description object which represents an application gateway in Azure
+   * @return an updated description object which represents an application gateway in Azure or null if the application gateway resource could not be retrieved
+   */
+  AzureAppGatewayDescription editAppGateway(AzureAppGatewayDescription appGatewayDescription) {
```

Comment:
Good point; the publicIpId was not set in the getDescriptionForAppGateway() but I don't see a reason why not setting that in there. I'll push a fix shortly.